src/main/java/tech/stackable/gis/hbase/BulkIngest.java

```diff
@@ -0,0 +1,95 @@
+package tech.stackable.gis.hbase;
+
+import ch.hsr.geohash.GeoHash;
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.*;
+
+import java.io.IOException;
+import java.util.PrimitiveIterator;
+import java.util.Random;
+
+public class BulkIngest {
+
+    private static final String usage = "ingest table source.tsv\n" +
+            "  help - print this message and exit.\n" +
+            "  table - the target table to load.\n" +
+            "  family - the column family to use.\n" +
+            "  count - the number of rows to generate and load.\n" +
+            "  limits - the applied co-ordinate ranage in the form: lon_min,lon_max,lat_min,lat_max\n" +
+            "\n" +
+            "generates a geohash for the rowkey.\n" +
+            "records are stored in columns in the specified family, columns are:\n" +
+            "  lon,lat,id,name,address,city,url,phone,type,zip\n";
+
+    public static void main(String[] args) throws IOException {
+
+        if (args.length != 4) {
+            System.out.println(usage);
+            System.exit(0);
+        }
+
+        final TableName tableName = TableName.valueOf(args[0]);
+        final Configuration conf = HBaseConfiguration.create();
+
+        HBaseAdmin.available(conf);
+        Connection connection = ConnectionFactory.createConnection(conf);
+        Table table = connection.getTable(tableName);
+
+        final Admin admin = connection.getAdmin();
+        byte[] family = String.valueOf(args[1]).getBytes();
+        int row_count = Integer.parseInt(args[2]);
+        long start = System.currentTimeMillis();
+        String[] limits = String.valueOf(args[3]).split(",");
+
+        // we need all 4
+        if (limits.length != 4) {
+            System.out.println(usage);
+            System.exit(0);
+        }
+
+        // generate random co-ordinates within a fixed region
+        PrimitiveIterator.OfDouble lon_iter = new Random().doubles(Double.valueOf(limits[0]), Double.valueOf(limits[1])).iterator();
```

Comment:
maybe a small sanity check that limits[0] < limits[1] and so on

Reply:
Done.